ckanext/harvest/harvesters/ckanharvester.py

```diff
@@ -236,24 +247,32 @@ def fetch_stage(self,harvest_object):
 
         # Get source URL
         url = harvest_object.source.url.rstrip('/')
-        url = url + self._get_rest_api_offset() + '/package/' + harvest_object.guid
+        url = url + self._get_action_api_offset() + '/package_show?id=' + harvest_object.guid
 
         # Get contents
         try:
             content = self._get_content(url)
+        except ContentNotFoundError,e:
+            # Remove package, as it no longer exists in the source:
+            self._remove_package({"id": harvest_object.guid})
+            harvest_object.report_status = 'deleted'
+            return True
```

Comment:
If you return here, will the report_status be saved?

Reply:
Yes, but better to be explicit. Adding save code.